tests/unit/directory_test.cc

```diff
@@ -74,13 +76,18 @@ int main(int ac, char** av) {
             });
         }
     };
+    fmt::print("--- Regular lister test ---\n");
+    return engine().open_directory(".").then([] (file f) {
+        return do_with(lister(std::move(f)), [] (lister& l) {
+          return l.done();
```

Comment:
The test is fairly bad since nothing guarantees it will test multiple calls into readdir.

Reply:
(but of course a preexisting problem)